### PR TITLE
DeltaLog.tableExists API

### PR DESCRIPTION
**Pull request by scottsand-db**
@scottsand-db Changed implementation to match Delta and check for snapshot version. If we just checked for `_delta_log` existence, (I assume) we would encounter the same error later on for an empty directory.

[Question] Are these tests sufficient or should it be tested for `/dir_a/` or `/dir_a/dir_b/_delta_log/dir_c` or `/dir_a/dir_b/_delta_log/` or something along those lines.
```
├── dir_a
│   └── dir_b
│         └── _delta_log
│                 └── dir_c
```

### CODE REVIEW

**Comment by scottsand-db** [comment]
> [Question] Are these tests sufficient or should it be tested for /dir_a/ or /dir_a/dir_b/_delta_log/dir_c or /dir_a/dir_b/_delta_log/ or something along those lines.

Does Delta OSS have any tests for similar tableExists APIs?

**Comment by allisonport-db** [comment]
> > [Question] Are these tests sufficient or should it be tested for /dir_a/ or /dir_a/dir_b/_delta_log/dir_c or /dir_a/dir_b/_delta_log/ or something along those lines.
> 
> Does Delta OSS have any tests for similar tableExists APIs?

No tests.

**Comment by scottsand-db** [comment]
I guess my question is what do you expect the answers to be for those cases you provided?
```
/dir_a/_delta_log --> true
/dir_a/dir_b/_delta_log/dir_c --> ? 
/dir_a/dir_b/_delta_log/  --> im assuming false? since dir_a isn't a delta table, dir_b is
```

If you want to manually compare against similar APIs in Delta OSS, that makes sense to me. Nonetheless having tests for the above examples ^ seems like a good idea to me.

**Comment by allisonport-db** [comment]
I mean I expect them all to be false. In `forTable` we do `apply(hadoopConf, new Path(dataPath, "_delta_log"))` only `/dir_a/dir_b/` should be true. I guess it might be worth just adding the tests.

**Comment by zsxwing** [comment]
An instance method is probably better. With the current API, the code would be

```
if (!DeltaLog.tableExists(hadoopConf, path)) {
  throw exception
} 
val deltaLog = DeltaLog.forPath(hadoopConf, path)
```

With an instance method, it can be
```
val deltaLog = DeltaLog.forPath(hadoopConf, path)
if (!deltaLog.tableExists()) {
  throw exception
}
```
The latter is cleaner as you don't need to pass `(hadoopConf, path)` twice. It also avoids creating DeltaLogImpl twice. Creating DeltaLogImpl is a heavy operation for big tables.

Delta has a similar API: https://github.com/delta-io/delta/blob/master/core/src/main/scala/io/delta/tables/DeltaTable.scala#L628-L647

But I think the current unit test in the PR is enough.

**Comment by scottsand-db** [comment]
You can add this under `// Public API changes in 0.2.0 -> 0.3.0` in StandaloneMimaExcludes

`ProblemFilters.exclude[ReversedMissingMethodProblem]("io.delta.standalone.DeltaLog.tableExists")`

File: standalone/src/main/scala/io/delta/standalone/internal/DeltaLogImpl.scala [hunk 1]
```diff
@@ -138,6 +138,9 @@ private[internal] class DeltaLogImpl private(
     new OptimisticTransactionImpl(this, snapshot)
   }
 
+  /** Whether a Delta table exists at this directory. */
+  def tableExists: Boolean = snapshot.version >= 0
```
Comment (zsxwing) [inline standalone/src/main/scala/io/delta/standalone/internal/DeltaLogImpl.scala:142]: nit: override

**Review by zsxwing — APPROVED** [review]

**Pull request merged**